src/v/config/node_overrides.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "config/convert.h"
+#include "model/fundamental.h"
+#include "ssx/sformat.h"
+
+#include <yaml-cpp/yaml.h>
+
+#include <iosfwd>
+#include <vector>
+
+namespace config {
+
+struct node_id_override {
+    node_id_override() = default;
+    node_id_override(
+      model::node_uuid key,
+      model::node_uuid uuid_value,
+      model::node_id id_value)
+      : key(key)
+      , uuid(uuid_value)
+      , id(id_value) {}
+    model::node_uuid key;
+    model::node_uuid uuid;
+    model::node_id id;
+
+private:
+    friend std::ostream&
+    operator<<(std::ostream& os, const node_id_override& v);
+    friend std::istream& operator>>(std::istream& is, node_id_override& v);
+    friend bool operator==(const node_id_override&, const node_id_override&)
+      = default;
+};
+
+struct node_override_store {
+    node_override_store() = default;
+    void maybe_set_overrides(
```

Comment:
eh, I'm ambivalent about this. i don't see an obvious need for it, since applying the overrides happens elsewhere, conditionally with respect to several other checks.